From 84221b0370a8b49e0fa2dbd88c89fb6e87ec95bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Luttringer?= <seblu@seblu.net>
Date: Fri, 24 Jan 2020 17:07:11 +0100
Subject: [PATCH] Revert "Don't depend on the execute bit to determine pie
 executable or shared object."

This reverts commit d653309de04ed10fdeda79f2c6ca7a7e96e122f1.
---
 magic/Magdir/elf | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/magic/Magdir/elf b/magic/Magdir/elf
index b5a2e00b..a35b615d 100644
--- a/magic/Magdir/elf
+++ b/magic/Magdir/elf
@@ -50,8 +50,9 @@
 !:mime	application/x-object
 >16	leshort		2		executable,
 !:mime	application/x-executable
->16	leshort		3		shared object,
-!:mime	application/x-sharedlib
+>16	leshort		3		${x?pie executable:shared object},
+
+!:mime	application/x-${x?pie-executable:sharedlib}
 >16	leshort		4		core file,
 !:mime	application/x-coredump
 # OS-specific
-- 
2.25.0

